### PR TITLE
Add static body creation equivalent for all collision shape sibling creations

### LINKED ISSUE #86626: Can't create static body for something else then trimesh collisions
### Tested versions

- Reproducible in : 3.5.3 4.x ( 4.1, 4.3.dev_build[13a0d6e9b253654f5cc2a44f3d0b3cae10440443], assuming all other version)

### System information

macOS 13.3.0 - Vulkan (Forward+) - integrated Apple M1 Pro - Apple M1 Pro (8 Threads)

### Issue description

Tere is no available possibilities for creating a static body for something else then trimesh collisions

### Steps to reproduce

Select a MeshInstance3D 
click on the Mesh drop down button on the top of the 3D viewport 
see there is no available options for creating a static body for something else then trimesh collisions

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by ajreckof**
Fix #86626

Just adds an option in the popup menu and handle it the same way it was handled for trimesh collision shapes

### CODE REVIEW

File: editor/plugins/mesh_instance_3d_editor_plugin.h [hunk 1]
```diff
@@ -69,6 +72,7 @@ class MeshInstance3DEditor : public Control {
 	Control *debug_uv = nullptr;
 	Vector<Vector2> uv_lines;
 
+	Vector<Ref<Shape3D>> create_shape_from_mesh(Ref<Mesh> mesh, int p_option, bool verbose);
```
Comment (AThousandShips) [inline editor/plugins/mesh_instance_3d_editor_plugin.h:75]:
```suggestion
	Vector<Ref<Shape3D>> create_shape_from_mesh(Ref<Mesh> p_mesh, int p_option, bool p_verbose);
```
Prefix arguments with `p_`

File: editor/plugins/mesh_instance_3d_editor_plugin.cpp [hunk 2]
```diff
@@ -58,6 +58,59 @@ void MeshInstance3DEditor::edit(MeshInstance3D *p_mesh) {
...
 }
 
+Vector<Ref<Shape3D>> MeshInstance3DEditor::create_shape_from_mesh(Ref<Mesh> mesh, int p_option, bool verbose) {
+	Vector<Ref<Shape3D>> shapes = Vector<Ref<Shape3D>>();
+	switch (p_option) {
+		case MENU_OPTION_CREATE_TRIMESH_STATIC_BODY:
+		case MENU_OPTION_CREATE_TRIMESH_COLLISION_SHAPE: {
+			shapes.push_back(mesh->create_trimesh_shape());
+
+			if (verbose && !shapes.size()) {
```
Comment (AThousandShips) [inline editor/plugins/mesh_instance_3d_editor_plugin.cpp:68]:
```suggestion
			if (verbose && shapes.is_empty()) {
```
Same for all

File: editor/plugins/mesh_instance_3d_editor_plugin.cpp [hunk 3]
```diff
@@ -58,6 +58,59 @@ void MeshInstance3DEditor::edit(MeshInstance3D *p_mesh) {
...
+				err_dialog->set_text(TTR("Couldn't create a Trimesh collision shapes."));
```
Comment (AThousandShips) [inline editor/plugins/mesh_instance_3d_editor_plugin.cpp:69]:
```suggestion
				err_dialog->set_text(TTR("Couldn't create a Trimesh collision shape."));
```

File: editor/plugins/mesh_instance_3d_editor_plugin.cpp [hunk 4]
```diff
@@ -58,6 +58,59 @@ void MeshInstance3DEditor::edit(MeshInstance3D *p_mesh) {
...
+				err_dialog->popup_centered();
+			}
+		} break;
+
+		case MENU_OPTION_CREATE_SINGLE_CONVEX_STATIC_BODY:
+		case MENU_OPTION_CREATE_SINGLE_CONVEX_COLLISION_SHAPE: {
+			shapes.push_back(mesh->create_convex_shape(true, false));
+
+			if (verbose && !shapes.size()) {
+				err_dialog->set_text(TTR("Couldn't create a single collision shapes."));
```
Comment (AThousandShips) [inline editor/plugins/mesh_instance_3d_editor_plugin.cpp:79]:
```suggestion
				err_dialog->set_text(TTR("Couldn't create a single collision shape."));
```

File: editor/plugins/mesh_instance_3d_editor_plugin.cpp [hunk 5]
```diff
@@ -58,6 +58,59 @@ void MeshInstance3DEditor::edit(MeshInstance3D *p_mesh) {
...
+				err_dialog->popup_centered();
+			}
+		} break;
+
+		case MENU_OPTION_CREATE_SIMPLIFIED_CONVEX_STATIC_BODY:
+		case MENU_OPTION_CREATE_SIMPLIFIED_CONVEX_COLLISION_SHAPE: {
+			shapes.push_back(mesh->create_convex_shape(true, true));
+
+			if (verbose && !shapes.size()) {
+				err_dialog->set_text(TTR("Couldn't create a simplified collision shapes."));
```
Comment (AThousandShips) [inline editor/plugins/mesh_instance_3d_editor_plugin.cpp:89]:
```suggestion
				err_dialog->set_text(TTR("Couldn't create a simplified collision shape."));
```

File: editor/plugins/mesh_instance_3d_editor_plugin.cpp [hunk 6]
```diff
@@ -67,41 +120,38 @@ void MeshInstance3DEditor::_menu_option(int p_option) {
...
+					ur->create_action(TTR("Create Trimesh Static Body"));
+				} break;
+				case MENU_OPTION_CREATE_SINGLE_CONVEX_STATIC_BODY: {
+					ur->create_action(TTR("Create Single Convex Static Body"));
+				} break;
+				case MENU_OPTION_CREATE_SIMPLIFIED_CONVEX_STATIC_BODY: {
+					ur->create_action(TTR("Create Simplified Convex Static Body"));
+				} break;
+				case MENU_OPTION_CREATE_MULTIPLE_CONVEX_CSTATIC_BODY: {
+					ur->create_action(TTR("Create Multiple Convex SStatic Body"));
```
Comment (AThousandShips) [inline editor/plugins/mesh_instance_3d_editor_plugin.cpp:141]:
```suggestion
					ur->create_action(TTR("Create Multiple Convex Static Body"));
```

File: editor/plugins/mesh_instance_3d_editor_plugin.h [hunk 7]
```diff
@@ -43,7 +43,10 @@ class MeshInstance3DEditor : public Control {
 	GDCLASS(MeshInstance3DEditor, Control);
 
 	enum Menu {
-		MENU_OPTION_CREATE_STATIC_TRIMESH_BODY,
+		MENU_OPTION_CREATE_TRIMESH_STATIC_BODY,
+		MENU_OPTION_CREATE_SINGLE_CONVEX_STATIC_BODY,
+		MENU_OPTION_CREATE_SIMPLIFIED_CONVEX_STATIC_BODY,
+		MENU_OPTION_CREATE_MULTIPLE_CONVEX_CSTATIC_BODY,
```
Comment (AThousandShips) [inline editor/plugins/mesh_instance_3d_editor_plugin.h:49]:
```suggestion
		MENU_OPTION_CREATE_MULTIPLE_CONVEX_STATIC_BODY,
```
Assuming this is a typo

**Comment by smix8** [comment]
The duplicated menu option names need to be fixed.

menu

Imo instead of making the menu even more convoluted than it already is it should be moved to a submenu where users can change the currently hardcoded settings.

**Comment by ajreckof** [comment]
I reworked it so that it is now in a submenu. On the side while reworking I ended up adding the posibility to add colision shape as sibling on a multi-selection too as it was previously only for static body child.

**Comment by smix8** [comment]
I tested with a the build artifact from this pr and a simple 2x2x2x BoxMesh.

The "Static Body Child" options all work as expected and create StaticBody3D child node with a CollisionShape3D child node that has a ConvexPolygonShape3D resource.

Now what does not seem to work as expected are the "Sibling" options.

I would expect them to do the very same except for leaving out the StaticBody3D node and place the CollisionShape3D node as a sibling with the same ConvexPolygonShape3D resource as with the other options.

The node creation and placement works but not the Shape3D resource creation. Not sure why but the sibling options all created ConcavePolygonShape3D for everything no matter what convex menu option is picked. I would expect them to create the same shape as the other convex options that work.

File: editor/plugins/mesh_instance_3d_editor_plugin.cpp [hunk 8]
```diff
@@ -58,200 +58,159 @@ void MeshInstance3DEditor::edit(MeshInstance3D *p_mesh) {
...
+	}
+	return shapes;
+}
 
-			CollisionShape3D *cshape = memnew(CollisionShape3D);
-			cshape->set_shape(shape);
-			cshape->set_transform(node->get_transform());
+void MeshInstance3DEditor::_create_collision_shape() {
+	int placement_option = shape_placement->get_selected();
+	int shape_type_option = shape_placement->get_selected();
```
Comment (ajreckof) [inline editor/plugins/mesh_instance_3d_editor_plugin.cpp:113]: this is the culprit code I used in both case `shape_placement` while the second one was supposed to be `shape_type` must have been a bad copy paste. :(

**Review by smix8 — APPROVED** [review]
Seem to work as expected now. I did not encounter any more issues.

**Review by AThousandShips — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**